src/dist/replication/lib/replica_bulk_load.cpp

```diff
@@ -71,7 +73,115 @@ void replica::on_bulk_load(const bulk_load_request &request, /*out*/ bulk_load_r
 // ThreadPool: THREAD_POOL_REPLICATION
 void replica::broadcast_group_bulk_load(const bulk_load_request &meta_req)
 {
+    if (!_primary_states.learners.empty()) {
+        dwarn_replica("has learners, skip broadcast group bulk load request");
+        return;
+    }
+
+    if (_primary_states.group_bulk_load_pending_replies.size() > 0) {
+        dwarn_replica("{} group bulk_load replies are still pending, cancel it firstly",
+                      static_cast<int>(_primary_states.group_bulk_load_pending_replies.size()));
+        for (const auto &kv : _primary_states.group_bulk_load_pending_replies) {
+            kv.second->cancel(true);
+        }
+        _primary_states.group_bulk_load_pending_replies.clear();
```

Comment:
Does it mean we will resend group_bulk_load_request to sendaries? How does secondary replica deal with duplicate bulk load request?

Reply:
- meta -> primary: send bulk _load_request
- primary -> secondary: send group_bulk_load_request (request 1)
- secondary -> primary : report bulk load state
- primary -> meta: report group bulk load states
- if bulk load not finished, meta send bulk_load_request again
- primary -> secondary: send group_bulk_load_request (request 2)

Primary will not resend group_bulk_load_request, request 1 and request 2 are different.
When primary tries to send request 2, primary will check if there are request 1 pending firstly, if there are pending request 1, primary will cancel tasks and send request 2.
Meta server will not send bulk_load_request too soon, the interval is longer than 5 seconds (will be shown in further pull request), it is safe to cancel the long-time, pending request.